force_bdss/data_sources/base_data_source_model.py

```diff
@@ -46,43 +47,187 @@ class BaseDataSourceModel(BaseModel):
     #: this and adapt the visual entries.
     changes_slots = Event()
 
-    def verify(self):
-        """ Verify the data source model.
+    def __init__(self, factory=None, input_slot_info=None,
+                 output_slot_info=None, *args, **kwargs):
+        super(BaseDataSourceModel, self).__init__(factory, *args, **kwargs)
 
-        The data source model must have:
-        - input and output slots match between instance and model
-        - all output slots named
-        - no errors in input or output slots
+        # If either input_slot_info or output_slot_info have been passed in
+        # as arguments, perform extra checks before assigning to make sure
+        # they will be accepted by the BaseDataSource associated with factory
+        if input_slot_info is not None:
+            self._assign_slot_info("input_slot_info", input_slot_info)
+        if output_slot_info is not None:
+            self._assign_slot_info("output_slot_info", output_slot_info)
 
-        Returns
-        -------
-        errors : list of VerifierErrors
-            The list of all detected errors in the data source model.
-        """
-        try:
-            data_source = self.factory.create_data_source()
-        except Exception:
-            logger.exception(
-                "Unable to create data source from factory '%s', plugin "
-                "'%s'. This might indicate a  programming error",
-                self.factory.id,
-                self.factory.plugin.id,
+    # -------------------
+    #      Defaults
+    # -------------------
+
+    def _input_slot_info_default(self):
```

Comment:
The default values for `input_slot_info` and `output_slot_info` are generated from the associated `BaseDataSource` of the `factory` attribute. Importantly, we assign the corresponding `type` and `description` attributes on each `InputSlotInfo` and `OutputSlotInfo`, since we can use these as references that should not be changed by the user.